languages/SMW_LanguageAr.php

```diff
@@ -52,62 +54,58 @@ class SMWLanguageAr extends SMWLanguage {
 
 	protected $m_SpecialProperties = array(
 		// always start upper-case
-		'_TYPE'  => 'لديه نوع',
-		'_URI'   => 'معرف الموارد الموحد معادلة',
-		'_SUBP' => 'الخاصية الفرعية ل',
-		'_SUBC' => 'تصنيف فرعي من',
-		'_UNIT' => 'عرض الوحدات',
-		'_IMPO' => 'المستوردة من',
-		'_CONV' => 'يقابل',
-		'_SERV' => 'يوفر الخدمة',
-		'_PVAL' => 'يسمح بالقيمة',
+		'_TYPE' => 'من النوع',
+		'_URI'  => 'URL مكافئ',
+		'_SUBP' => 'خصيصة فرعية من',
+		'_SUBC' => 'مُصنّف على التصنيف',
+		'_UNIT' => 'وحدة العرض',
+		'_IMPO' => 'مستوردة من',
+		'_CONV' => 'توافق',
+		'_SERV' => 'تقدّم خدمة',
+		'_PVAL' => 'تقبل القيمة',
 		'_MDAT' => 'تاريخ التعديل',
-		'_CDAT' => 'Creation date', // TODO: translate
-		'_NEWP' => 'Is a new page', // TODO: translate
-		'_LEDT' => 'Last editor is', // TODO: translate
-		'_ERRP' => 'يمتلك قيمة غير صحيحة ل',
-		'_LIST' => 'يمتلك حقول',
-		'_SOBJ' => 'Has subobject', // TODO: translate
-		'_ASK'  => 'Has query', // TODO: translate
-		'_ASKST'=> 'Query string', // TODO: translate
-		'_ASKFO'=> 'Query format', // TODO: translate
-		'_ASKSI'=> 'Query size', // TODO: translate
-		'_ASKDE'=> 'Query depth', // TODO: translate
-		'_ASKDU'=> 'Query duration', // TODO: translate
-		'_MEDIA'=> 'Media type',
-		'_MIME' => 'MIME type',
-		'_ERRC' => 'Has processing error',
-		'_ERRT' => 'Has processing error text',
-		'_PREC'  => 'Display precision of',
-		'_LCODE' => 'Language code',
-		'_TEXT'  => 'Text',
-		'_PDESC' => 'Has property description',
-		'_PVAP'  => 'Allows pattern',
-		'_DTITLE' => 'Display title of',
-		'_PVUC' => 'Has uniqueness constraint',
+		'_CDAT' => 'تاريخ الإنشاء',
+		'_NEWP' => 'هي صفحة جديدة',
+		'_LEDT' => 'آخر مَن حرّرها',
+		'_ERRP' => 'فيها قيمة غير صحيحة في',//الخصيصة
```

Comment:
Thanks for your review and feed back.

Let me first comment about the rationale: You probably know, Meno, that many norms and widely accepted conventions of localisation have been changed and re-thought during the past ten years of FOSS localisation. Including the introduction of idioms that hadn't existed before, and bringing others under critique. Whether those changes are for the better or for the worse. This will remain controversial. I personally think that much remains in need of rethinking. I also know that the Arabic Wikipedia community has become increasingly more conservative; not only in regard to localisation.

Regarding the short months names case: I know they are used in Gnome, a major FOSS package, as well as the Mozilla suite.

Language - and modes thereof - is just another technology of communication. It has phases. ICT has introduced modes and idioms that hadn't existed before, and I don't see why we should insist to sticking to what has been known before in this specific point, i.e. month names. For me if is is useful, it should be allowed; people should be given the chance to try them. After all in all applications I know of, dates formats are selectable. No imposing of anything is taking place here.

As for tashkeel: You probably know that it is an essential part of the Arabic writing system, without which it is not considered complete. The absence of tashkeel from computing has been traditionally - in my opinion - due to the difficulties of presentation; i.e fonts, resolutions, encoding. This is all mostly of the past now. What remains of these difficulties should be tackled technically, if possible, rather than imposing restrictions in how the language is used. The language is and will continue to change because of the effects of technology. I just don't see this change and rational as valid or necessary. An example is the search function in browsers, which should implement switches to ignore tashkeel, as what popular search engines are doing. in turn search engines SHOULD implement switches to not ignore tashkeel because this can make a hell lot of a difference in search results in some cases, such as classical texts. The Arabic keyboard is evil, but that's another discussion.

Most importantly, the absence of tashkeel causes ambiguity and/or prevents the use of concise expressions that are arguably more suitable in certain cases, for fear of ambiguity.

I personally think that since SMW is not yet an integral part of Wikipedia, and that SMW is almost not used in Arabic mediawikis at all, it should not be strictly governed by WP-ar conventions. At least not without a space for reconsidering solutions.

In all cases, for the sake of the projects I'm working on, I will have to maintain a different translation for this and some other modules, as we do for our base MW installations, and hope that new insights in the discussions will bring them closer in the future. Even though I doubt this will be possible, if the rationale is basically "we need to do it this way because this is how it has been done so far in those other places" due of the exponential difficulty of effecting change in the body of localisation texts, while in fact fixing a small part and rarely used module such as SMW should be independently possible; but alas.

Thank you all, and regards.